wordpress-mobile/WordPress-iOS
# Stats Revamp: Views Visitors Details Screen

This PR adds the Details (2nd level) Views / Visitors screen

Ref #18601

Note:
 * this PR does not handle the logic to re-request data when changing the date selection header at top
 * the date selection header does not fully match the design (it has an extra detail text line) as I've chosen to re-use the existing header for now from the Period screens
 * this PR does not add the PIeChart for Referrers
 * this PR does not have the Countries Header as pat of Stats Revamp (Countries hasn't been converted for Stats Revamp yet)
 * this PR does not handle Ghost cards
 * this PR does not handle dynamically showing the date picker on 2nd level based on the stat section type.  will be done in future PR

Design
![image](https://user-images.githubusercontent.com/88816724/168612198-f1434fd2-35d7-48e3-882e-4b9f14169658.png)


Implementation
![image](https://user-images.githubusercontent.com/88816724/168612315-11a93518-4040-4465-9a22-fabeec86bc4d.png)
![image](https://user-images.githubusercontent.com/88816724/168612343-33efe758-05c2-4f00-b6ca-769eaf41e186.png)


To test:
1.  First, build and run, navigate to Stats Insights for a site and ensure that there is no option to add a Views & Visitors card like in s/shot below2. 

2. Click around insights, period stats (days, months, years) and ensure there are no crashes
4. Navigate out of stats to posts, media etc, and back to stats again and insights, period stats etc. + ensure there is no issue.
5. Now enable both of the new stats feature flags and build and run again

```
statsNewAppearance
statsNewInsights
```

6. Add New Stats Card = Views & Visitors.  New Views & Visitors Card and Graph should be added to the screen
7. Tap "This week" on the upper right of the Views / Visitors card.  Screen will segue / navigate to the Details card like below

8. Scroll up / down the details  screen.  There shouldn't be anything amiss / crashing 


## Regression Notes
1. Potential unintended areas of impact
Stats screens (insights and periods)

3. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and added a unit test

4. What automated tests I added (or what prevented me from doing so)
I added a unit test for ImmuTableRow creation

PR submission checklist:
